config/templates/stack-template.json

```diff
@@ -123,11 +130,20 @@
           }
         ],
         "VPCZoneIdentifier": [
+          {{if $.Controller.TopologyPrivate}}
+          {{range $index, $controllerSubnet := .Controller.PrivateSubnets}}
+          {{with $controllerSubnetLogicalName := printf "Controller%s" $controllerSubnet.LogicalName}}
+          {{if gt $index 0}},{{end}}
+          {
+            "Ref": "{{$controllerSubnetLogicalName}}"
+          }
+          {{end}}
+          {{end}}
+          {{else}}
           {{range $index, $subnet := .Subnets}}
```

Comment:
can't we implicitly populate `Controller.PrivateSubnets` with "global" subnets if they are not set? Then there will be less branching all over templates